Session05/lesson.ipynb

```diff
@@ -0,0 +1,1709 @@
Ab37zm+IuopCWTiEenHRO8ZJOvh5qqihWjcfj4pRjui2DOg3DwLvvvouvf/3rcuDT7sv0XQYEUp9AR9bqM+VBsUoL6bouacm0yycSCQDA06dP8bWvfU1Sfr1eryTw8rXyQNB1Hfl8Hnt7e9ccRczjYkE3m83kMKHGLR6PS7YOdXC12j4ikYjQHEuK7EeCrnEDZiHPzRfANQ0H5wA7fxodtre3JbmaBSF/FtEtFmuMAeABRZqSdDZF4U6nE9VqVXJrVFWV4D3q6JgvxjweAJILRCqYtKfFYpGLRM/Pz7G7uyvp2KuJwE+ePJG0Yn6evNKHaBppUbpXKQpmkcHPi3Z2Upfvvvsu3n77bfk7RC+Zm0NnGPcaanVYtHP/YvYT9wYGPVLPw2gAXqWRzWal2D07O8PBwQGm06lcZMyDj4GjRNqIJiUSCVQqFTno2bAwm6xWq8lderyLku5AYNko7u7uCv1NFDwWi0lBTCS0Wq3i6OgIv/VbvyXPc1VDZBgGXrx4gTt37gCAaJ24vhncudRv/TrK5X+CO3fuiEt41TnKe9go3mYBwv2DDdOjR49w7949GIYhIm7SVEz4tlqtODo6wt7enuRuMVAXgLiGuRdTWsF9Z1X0TRpeURTRmREpokmChphnz55Jo0cJCD9P7rdcw3TEci0x8JNiazbiwFLWwFgYak35Ob333nu/snZ9RVH+IYD/GMB/D+D//okvF03TLF4VT38FIAvgv8CSMvuvAbwG4L5pmoWVn/dPAHzz6vtyWIZF/g6AL5um+eFnfZ0/b3ymgEfCjvV6HR999BGm0ym2t4vy9VKpJHqNWq2OePyVJH7ScUE+3Wq1Sujg7u6uiD0ZShaPxyXQbH9/Hx6PB4VCQUSH7J6Ze8LK/fT09OrCUYssIG50s9kMt2/fRiaTEbcH4VRqh6iBsdsdV7SEXTRIoVAImvZNCaMk/UWxHa/NsFgsgka8fPlSBKo8IFeFr4FAQKgUbg4UnAMQNCadTsu1EdTrPH/+HDs7OxI0x8wOQuNutxupVEoOCHafpIJisRgASKI4LwWljZmfKw9DRvvz2bIoarVaqNfraLVaeP311/9f9t6st/U8ze/7chFFUTtFcZGofTlbbe6arppebLgdeJxgBnkFhu0MBgZseG5ym5sOkCu/AF/kJoDfQC4MO8D0zPT0dFV39ant7No3ihRJLZQoSqI2krmgPt+mOp5JqoJBYZJDoNBdy5FI/n/L83y3R6enpxbckuFSq9U0OTlp8SdCSyzY6Ir47E+ePFEgEHAnjnOHrg3dFnlLFMWkzaJrQxDfmUr85MkTxWIxU3hYrXE0QWVg5SesU2ojKti5r6+vFYt9brSnp+epurs/MzXMvCsQIZoC1sLg4KBD63j+zI+Lx+P36M9kMnlPWMyoGArS7e1to1DhcFhra2va3t7WxsaGHTy4EaPRqJEchg+DRCG8Pj8/VzweNy3J90EwH9Ta8vKyurq6HLkBJQSaSLHZ39+vjz/+WOl02loyjAUUlGtra548fnFxYa0P4ZPNZtP0DnoNUuN//OMf+3KmqHv06JH3Wr1eNx3PXgVVRK+zu7vr4qlarerly5caHh62uBlqJJfLueAul8seZcJ3QHo1OhrQqHK5rPX1dZXL5fYhfFdQUtwSJUFcBxEBIFg7Ozv+7JIcwgkSND4+rtHRURtjisVFVSrvqlb7ngusH//4x/fGYbAuET3jgCSLDW0pIv2dnR2dnJyo2fy5pqamTGtzTm1tbWlgYECxWMwIHwgjTRcoczgc1j/6R/9I0Wg7KPb4+FhPnjyxVKAzowhUG7NNV1eX1+zIyIg1TsxGjEbbqeWErPb19bnRqdfr2t3ddQPY/q6KLqROT0+1s7OjnZ0dhcNhuxQprjijOLunpqa8jnjWkUhEuVzOsRx9fe2B2YThMiSc/x7d0t/z139397//k6Rf/85ffyJJrVarqXZx8zNJ/0HS/652WvZPOouiu9f/IOl/k/S/SPrPahdT/+3fVVEkfYvCCG0N1te5uTmdnZ3dBZ09dbbJ/v7+HbKwrKOjI0exE/YHTdEZTf/VV185kbZer2tlZcWwIkJXuljCDTtFkLiu2IypVMrc8PDwsLa2tqy/YDggHV+7u7mxvZnNh8NIatMxdC/A1ul02pc7LifyORAatlotvfvuu/fcbVdXv6/r6x8oFPon2tmZdhJ3J/VEJwQqw6BWOHzsnk+ePDGPjnW7VCq5+yOdlS4GFAZLfOf0dbq8Tg0CYWTkhFQqFVucuVhAfbLZrGZmZmyrxo1CthEuokgkor/6q7/ys2EWFMNP6b5yuZwGBgZsdeV3Pnr0yGsS59Xv/d7vuTCu1+tG/ihSt7a2rBtBF3B6euoCGmElhdX8/Lw1HpI892tjY8P6MTradkjnU/X2fmEaDariyy+/dDfN+iXfKpfLaX9/X4VCQalUynlfkUhEpVLJAn8O31qtZpQRpHVra8uXO46fXC5nMXMikXBSMujH2NiY0SKeM9QxAnqeweXlpalHxmDwvkB2Hj9+7GKGf0bWD2tFksfRcNnSYXd1denrr7+2w5JiLRAIaGVlxTTH5eXlnd2+Yks3cSHb29t2C1JcckmXSiUjfFBMwWDQCC+FayqV0uLiora2tmx///jjj42MsXZAUCj+pHacBbO10BJ2d3drenraKfrn5+e6vb3Vw4cP7ciEis3n80ZCyAYDVZXazdGjR4+UTCZtWmE22cTEhAKBgF6+fOnQSkk6O2ubijjrKDLJWiOrLRAIqKenR2tray4maTg7w0FZJ5OTk6Zk+Zw0lJ0p+ojgS6WS5yGiAyRiYn9/383H0dGRZmZmLK/gufNMp6en3QBiTqG4+/zzz/1ZpHZmEkOpaeLIt+rMsGK8DGcs44JAcN977z3d3t7q9evXqtVqSiaTRimREjSbTe9NtH2cvzMzMw6qhe4HIUKDBYKP8P/v86vVak23Wq3A3/DXTzv+u0qr1frjVqsVb7VasVar9d+
```

Comment:
Consider using a for loop to build these objects, its a ton of visual noise to read this cell
something like:
```python
objects = []
for filter in ("HSC-I", "HSC-R", "HSC-G", "HSC-Z"):
    objects.append(butler.get("deepCoadd_forced_src, dataid={'filter':filter, 'field':field, 'tract':15830, 'patch':'0,3'})
iSources, rSources, gSources, zSources = objects
```
or directly:
```python
dataId = {'filter':None, 'field':field', 'tract':15830, 'patch':'0,3'}
iSources, rSources, gSources, zSources =\
    (butler.get('deepCoadd_forced_src' dataid=dataId.update({'filter': f})
    for f in ("HSC-I", "HSC-R", "HSC-G", "HSC-Z"))
```
It is fine as it is, and these examples may bring up question un related to what you are trying to show, but are also visually distracting. Maybe even just factoring out the strings and dataId would be enough. Same in the example below.